handles.go

```diff
@@ -52,12 +52,12 @@ func (v *HandleList) Track(pointer interface{}) unsafe.Pointer {
 
 	v.Unlock()
 
-	return unsafe.Pointer(&slot)
+	return unsafe.Pointer(uintptr(slot))
```

Comment:
This changes what we're doing, doesn't it? We were taking its address and making it a `unsafe.Pointer`. In the new code we're taking the slot value and casting it a few times. Should this still work?